### PR TITLE
ramips: enable LED button for TP-Link EC330-G5u v1

### PR DESCRIPTION
**Pull request by csharper2005**
The device already has LED push button (`KEY_LIGHTS_TOGGLE`) and exported GPIO control `led-light`. This commit adds button handler script for switching on/off all device LEDs.

### CODE REVIEW

**Comment by chunkeey** [comment]
Are you missing another patch? Or is there a dependency with another PR?
(I'm asking because I can't see the KEY_LIGHTS_TOGGLE in button-hotplug, so I'm wondering how this works)

**Comment by csharper2005** [comment]
@chunkeey this simply works. Without another patches and dependencies.

I found a hint in the OpenWrt wiki https://openwrt.org/docs/guide-user/hardware/hardware.button:
_Please consider using these kernel codes when adding support for new devices, they're mapped by the gpio-button-hotplug kernel module:_

<html>
<body>


021e | KEY_LIGHTS_TOGGLE | lights_toggle
-- | -- | --



</body>
</html>

I found this after your question (maybe, this explains how it works):
https://github.com/openwrt/openwrt/blob/21f713d5abf86fc5639b41d7f4e7535a5538d63a/package/kernel/gpio-button-hotplug/src/gpio-button-hotplug.c#L91
https://github.com/openwrt/openwrt/blob/21f713d5abf86fc5639b41d7f4e7535a5538d63a/package/system/procd/files/hotplug.json#L72

**Comment by chunkeey** [comment]
you are right, I was looking in button-hotplug and not gpio-button-hotplug. My bad
(maybe button-hotplug needs an update... but this is not a concern for your patch).

File: target/linux/ramips/mt7621/base-files/etc/rc.button/lights_toggle [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+[ "${ACTION}" = "released" ] || exit 0
+
+. /lib/functions.sh
+
+case $(board_name) in
+	tplink,ec330-g5u-v1)
+		led_light="/sys/class/gpio/led-light/value"
+		if [[ $(cat $led_light) == 0 ]]; then
```
Comment (obsy) [inline target/linux/ramips/mt7621/base-files/etc/rc.button/lights_toggle:10]: Bashism. Use 
```
if [ $(cat $led_light) = 0 ]; then
```

Comment (csharper2005) [reply target/linux/ramips/mt7621/base-files/etc/rc.button/lights_toggle]: Thanks. Done.

File: target/linux/ramips/mt7621/base-files/etc/rc.button/lights_toggle [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+. /lib/functions.sh
+
+case $(board_name) in
+	tplink,ec330-g5u-v1)
+		led_light="/sys/class/gpio/led-light/value"
+		if [ $(cat $led_light) = 0 ]; then
+			echo 1 > $led_light
+		else
+			echo 0 > $led_light
```
Comment (DragonBluep) [inline target/linux/ramips/mt7621/base-files/etc/rc.button/lights_toggle:13]: It can be simplified to `echo $(($(cat $led_light) ^ 1)) > $led_light`.

Comment (csharper2005) [reply target/linux/ramips/mt7621/base-files/etc/rc.button/lights_toggle]: Thanks. I like this. Done.

**Comment by chunkeey** [comment]
Merged. Thanks!

**Pull request closed without merging**